usr/src/cmd/zoneadmd/log.c

```diff
@@ -691,83 +737,101 @@ logfile_write(logfile_t *lfp, const char *buf, size_t buflen)
 	}
 }
 
+static void
+add_bunyan_preamble(custr_t *cus)
+{
+	struct tm gtm;
+	struct timeval tv;
+	/* Large enough for YYYY-MM-DDTHH:MM:SS.000000000Z + NUL */
+	char timestr[32] = { 0 };
+	size_t len;
+
+	if (gettimeofday(&tv, NULL) != 0 ||
+	    gmtime_r(&tv.tv_sec, &gtm) == NULL) {
+		logstream_err(B_TRUE, "failed to get time of day");
+		abort();
+	}
+
+	len = strftime(timestr, sizeof (timestr) - 1, "%FT%T", &gtm);
+	VERIFY3U(len, >, 0);
+	VERIFY3U(len, <, sizeof (timestr) - 1);
+
+	VERIFY0(custr_append_printf(cus, "\"time\": \"%s.%09ldZ\", ",
+	    timestr, tv.tv_usec * 1000));
+	VERIFY0(custr_append_printf(cus, "\"v\": %d, ", BUNYAN_VERSION));
+	VERIFY0(custr_append_printf(cus, "\"hostname\": \"%s\", ", host));
+	VERIFY0(custr_append(cus, "\"name\": \"zoneadmd\","));
+	VERIFY0(custr_append_printf(cus, "\"pid\": %s, ", pidstr));
+	VERIFY0(custr_append_printf(cus, "\"level\": %d", BUNYAN_LOG_LEVEL));
+}
+
 /*
  * Convert the json pairs into a json object.  A "time" element is added to
```

Comment:
can we drop the outdated "A time element" sentence

Reply:
Done